src/main/java/org/opensearch/ad/ml/EntityModel.java

```diff
@@ -47,6 +49,19 @@ public EntityModel(Entity entity, Queue<double[]> samples, RandomCutForest rcf,
         this.threshold = threshold;
     }
 
+    /**
+     * Constructor with ERCF.
+     *
+     * @param entity entity if any
+     * @param samples samples with the model
+     * @param ercf extended rcf model
+     */
+    public EntityModel(Entity entity, Queue<double[]> samples, ExtendedRandomCutForest ercf) {
+        this.entity = entity;
+        this.samples = samples;
+        this.ercf = ercf;
+    }
```

Comment:
Are `this.rcf` and `this.threshold` initialized to `null`? Since ERCF = RCF + Thresholder it's important that this `EntityModel` doesn't have instances of all three types.

Reply:
yes, default values are null.